fvm/environment/crypto_library.go

```diff
@@ -0,0 +1,106 @@
+package environment
```

Comment:
Would be awesome to add spans to the public functions here since they may be expensive.  Or just leave TODOs for me.